### PR TITLE
[codex] Improve Kiwi waterfall clarity and auto scaling

### PR DESCRIPTION
**Pull request by rfoust**
## Summary

Improves KiwiSDR waterfall clarity and control behavior by moving Kiwi W/F rows into the 2D waterfall history/texture path instead of treating each update as a stretched 1D row. Each decoded dBm row is stored with its capture frequency frame and remapped into the current panadapter view, using peak-preserving sampling so narrow carriers survive zoom/span changes. GPU builds render that 2D waterfall texture through the existing QRhi path, while the QImage/non-GPU fallback remains intact.

This also replaces the previous gain/black-level interpretation with Kiwi-style WF Floor / WF Ceiling scaling. The Kiwi Auto path now uses a clean-room percentile algorithm with temporal/spatial smoothing, preserves live range state while Auto is pending, keeps manual slider edits anchored to the current live range, normalizes stored Auto ranges across zoom changes, and makes user-requested Auto publish the fresh calculated range when enough rows are available. The automation bridge also gains pan-scoped targeting so these controls can be tested reliably when multiple panadapters or hidden duplicate menus exist.

## Screenshots

Captured from the final local build against K4JK.

Kiwi Auto waterfall scale:

Kiwi Auto waterfall scale

Manual anchored WF Floor / WF Ceiling:

Manual anchored WF Floor and WF Ceiling

Auto restored:

Auto restored

## Constitution principle honored

Principle IV — the KiwiSDR protocol/display behavior is implemented from clean-room, source-attributed protocol facts and user-provided requirements, with no KiwiSDR code copied, translated, vendored, or mechanically ported.

## Test plan

- [x] Local build passes (`cmake --build build --target AetherSDR --parallel`)
- [x] Behavior verified on a real radio if applicable
  - Launched with `AETHER_AUTOMATION=1`
  - Connected to the local FLEX-6700
  - Assigned slice 0 to the saved K4JK KiwiSDR profile
  - Verified observed direct Kiwi W/F rows through `get kiwi`
  - Opened pan 0 Display via the automation bridge
  - Captured PR screenshots for Auto, manual anchored floor/ceiling, and restored Auto states
  - Verified WF Ceiling edits keep the current WF Floor anchored
  - Verified WF Floor edits keep the current WF Ceiling anchored
  - Verified Auto returns to checked and recalculates live values
  - Restored K4JK saved profile values
  - Restored slice 0 back to Flex receive source
- [x] Existing tests pass locally
  - `cmake --build build --target kiwi_sdr_protocol_test --parallel`
  - `./build/kiwi_sdr_protocol_test`
  - `git diff --check`
- [ ] Existing tests pass (CI)
- [x] Reproduction steps documented if user-reported bug
  - User-reported issue was blurry/overstretched Kiwi waterfall rendering and unstable/incorrect Auto floor/ceiling behavior compared with Flex waterfall clarity.

## Checklist

- [x] Commits are signed (`docs/COMMIT-SIGNING.md`)
- [x] No new flat-key `AppSettings` calls — use nested-JSON-under-one-key
      (Principle V)
- [x] Code is clean-room — not decompiled, disassembled, or
      reverse-engineered from a proprietary binary (Principle IV)
- [x] All meter UI uses `MeterSmoother` (AGENTS.md convention)
- [x] Documentation updated if user-visible behavior changed
- [x] Security-sensitive changes reference a GHSA if applicable

### CODE REVIEW

**Review by ten9876 — COMMENTED** [review]
Thorough, well-built feature — the move to a 2D history/texture path and a clean-room percentile Auto is the right direction, and I appreciate the hardware verification against K4JK. I ran a full multi-angle review; the happy path holds up, but the new Auto-scale has a cluster of edge-case correctness gaps and the settings migration has a real upgrade issue. Inline comments carry the specifics; summary below.

**Would like fixed before merge** (inline): (1) Auto floor/ceiling bypass the `[-260, 30]` clamp the manual path enforces; (2) the legacy-settings migration silently re-baselines and disables Auto for existing users who touched the sliders; (3) a `wf_cal` change mixes differently-calibrated rows in the 20-row Auto window; (4) the band-change Auto reset is skipped when the prior band name was empty.

**Polish** (inline): (5) `waterfallByteToDbm` dropped its `[-200,0]` clamp but the FFT-trace/squelch consumers still floor at −200; (6) tracking-only-fallback profiles never get the Auto range pushed to the sliders/pan; (7) the WF Floor/Ceiling slider label reads "dB" for values that are now absolute dBm.

**User-facing changes worth a conscious ack** (not bugs): the color curve went `sqrt` → linear (intentional, now documented in the design doc §857-858); the default range changed `-130/-50` → `-110/-10`; the shared auto-black button now triggers auto-scale; and `doInvoke` now refuses invisible targets (this is the mechanism for the "hidden duplicate menu" targeting the PR adds — intended, but confirm no existing automation drove controls on inactive tabs).

**Non-blocking cleanup/perf I noticed:** `KiwiSdrClient::decodeWaterfallFrame()` is now a dead member (no callers; the decode+calibration is re-inlined at the call site); `pushKiwiSdrWaterfallRow` recomputes the default range + scheme stops per bin and heap-allocs two `QVector<QRgb>` per row (hot path — hoist per-row); `autoWaterfallDisplayRange` full-sorts 1024 bins for two order statistics (`std::nth_element`); the dBm limit constants are duplicated across 4 files; the repurposed WF sliders don't update `setAccessibleName/Description` (only the Auto button got it, per docs/a11y.md); the `512`/`1003` percentile anchors are uncommented; and the manual-set path double-emits `waterfallDisplayRangeChanged` (deduped downstream, harmless).

Genuinely nice work — findings 1–4 are the ones I'd want addressed before this lands. 73 — reviewed with Claude Code on behalf of KK7GWY.

File: src/core/KiwiSdrClient.cpp [hunk 1]
```diff
@@ -1438,21 +1507,159 @@ void KiwiSdrClient::sendWaterfallViewToServer()
...
+    if (!m_waterfallAutoScaleEnabled) {
+        return KiwiSdrProtocol::adjustedWaterfallDisplayRange(
+            static_cast<float>(m_waterfallManualMinDbm),
+            static_cast<float>(m_waterfallManualMaxDbm),
+            0,
+            0);
+    }
+
+    if (m_waterfallClientAutoRangeValid) {
+        return adjustedAutoWaterfallDisplayRange(
```
Comment (ten9876) [inline src/core/KiwiSdrClient.cpp:1567]: **Auto range isn't clamped to the dBm limits.** The manual path clamps to `[kWaterfallMinDbmLimit, kWaterfallMaxDbmLimit]` in `setWaterfallDisplayRange`, but the Auto branch returns `adjustedAutoWaterfallDisplayRange(...)`, and neither that nor `KiwiSdrProtocol::adjustedWaterfallDisplayRange` clamps to the limits (only `isfinite` + a min-span guard). The Auto min/max come straight from `sorted[512]-10` / `sorted[1003]+30`, so with a positive `wf_cal` and a strong carrier the ceiling exceeds +30, and on a dead-quiet band the floor drops below −260. That gets sent to the server as an out-of-range `SET maxdb/mindb`, and emitted to the GUI where the slider clamps to the limit while `SpectrumWidget` color-maps against the raw value — so the slider position no longer matches the actual waterfall floor. Suggest clamping `range.minDbm`/`range.maxDbm` to `[kWaterfallMinDbmLimit, kWaterfallMaxDbmLimit]` here (or where the client auto range is stored in `updateWaterfallAutoScale`), matching the manual path. (#4069 review)

File: src/core/KiwiSdrManager.cpp [hunk 2]
```diff
@@ -801,10 +870,36 @@ void KiwiSdrManager::loadSettings()
...
+                kKiwiSdrDefaultWaterfallMinDbm + legacyMinOffsetDb,
+                kKiwiSdrWaterfallMinDbmLimit,
+                kKiwiSdrWaterfallMaxDbmLimit - 1);
+            p.waterfallMaxDbm = std::clamp(
+                kKiwiSdrDefaultWaterfallMaxDbm + legacyMaxOffsetDb,
+                p.waterfallMinDbm + 1,
+                kKiwiSdrWaterfallMaxDbmLimit);
+            p.waterfallAutoScale =
+                legacyMaxOffsetDb == 0 && legacyMinOffsetDb == 0;
+        }
```
Comment (ten9876) [inline src/core/KiwiSdrManager.cpp:894]: **Legacy migration silently re-baselines and disables Auto.** The old `waterfallCellDb`/`waterfallFloorDb` were ±dB offsets against the *old* −130/−50 baseline, applied on top of a continuously-running auto range. This adds them onto the *new* −110/−10 baseline (so a customized ceiling jumps up to +40 dB) **and** sets `autoScale=false` whenever either offset was non-zero — so any existing user who ever nudged WF Gain/Floor upgrades to a washed-out, non-auto waterfall on first launch. Since the old cell/floor semantics have no faithful mapping onto the new absolute WF Floor/Ceiling model, the cleanest migration is to reset legacy profiles to Auto:

```suggestion
        } else {
            // Legacy profiles stored waterfallCellDb/waterfallFloorDb as ±dB
            // offsets on the OLD -130/-50 baseline with a continuously running
            // auto range. Those semantics don't map onto the new absolute WF
            // Floor/Ceiling model, so rather than re-baseline onto -110/-10
            // (which shifts a customized ceiling up to +40 dB) and pin
            // autoScale off, migrate legacy profiles to Auto. (#4069 review)
            p.waterfallMinDbm = kKiwiSdrDefaultWaterfallMinDbm;
            p.waterfallMaxDbm = kKiwiSdrDefaultWaterfallMaxDbm;
            p.waterfallAutoScale = true;
        }
```

File: src/core/KiwiSdrClient.cpp [hunk 3]
```diff
@@ -1438,21 +1507,159 @@ void KiwiSdrClient::sendWaterfallViewToServer()
...
+    emit waterfallDisplayRangeChanged(range.minDbm, range.maxDbm, autoRange);
+}
+
+void KiwiSdrClient::recordWaterfallAutoScaleRow(const QVector<float>& binsDbm)
+{
+    if (binsDbm.size() != kDefaultWaterfallFftBins) {
+        return;
+    }
+
+    m_waterfallAutoScaleRows.append(binsDbm);
```
Comment (ten9876) [inline src/core/KiwiSdrClient.cpp:1622]: **Calibration changes mix differently-calibrated rows in the Auto window.** Rows are stored here already offset by the *current* `m_waterfallCalibrationDb`. When the server sends `MSG wf_cal=<dB>` — which commonly arrives *after* the first waterfall rows on connect, and can change at any time — the up-to-19 rows already in `m_waterfallAutoScaleRows` were calibrated with the old offset, so `averageWaterfallRows()` blends rows that differ by the full calibration delta and the computed Auto floor/ceiling is skewed for ~20 frames. Clearing `m_waterfallAutoScaleRows` (and re-arming pending) wherever `m_waterfallCalibrationDb` is updated would keep the window internally consistent. (#4069 review)

File: src/core/KiwiSdrClient.cpp [hunk 4]
```diff
@@ -935,23 +947,34 @@ void KiwiSdrClient::disconnectFromEndpoint()
...
         && m_trackedMode == mode
         && m_trackedFilterLowHz == filterLowHz
         && m_trackedFilterHighHz == filterHighHz
-        && m_trackedPanId == panId) {
+        && m_trackedPanId == panId
+        && m_trackedBandName == normalizedBandName) {
         return;
     }
 
+    const bool bandChanged = !m_trackedBandName.isEmpty()
```
Comment (ten9876) [inline src/core/KiwiSdrClient.cpp]: **Band-change Auto reset is skipped when the prior band was unknown.** `bandChanged` requires `!m_trackedBandName.isEmpty()`, so tuning from a frequency where `BandSettings::bandForFrequency()` returned an empty name (out-of-band / unknown) into a real band yields `bandChanged=false` — the stale Auto range from the previous location stays applied until the user manually clicks Auto. Consider also treating an empty→non-empty transition as a change (or gating the reset on a significant frequency delta rather than the band-name string), so an Auto range computed on an unknown band doesn't persist across a retune. (#4069 review)

File: src/core/KiwiSdrProtocol.cpp [hunk 5]
```diff
@@ -427,12 +492,23 @@ quint64 sequenceGapCount(int previousSequence, int currentSequence)
     return delta > 0 ? static_cast<quint64>(delta - 1) : 0;
 }
 
+float waterfallByteToDbm(unsigned char value)
+{
+    return static_cast<float>(value) - 255.0f;
```
Comment (ten9876) [inline src/core/KiwiSdrProtocol.cpp:497]: The old `waterfallByteToDisplayLevel` clamped to `[-200.0f, 0.0f]`; `waterfallByteToDbm` now returns the raw `value - 255` (down to −255 dBm). That's a deliberate range expansion, but several `SpectrumWidget` consumers of the same decoded bins still floor at `kKiwiSdrWaterfallMinDbm = -200` (`estimateKiwiSdrTraceFloorDbm`, `reprojectBins`, the squelch visual floor, `peakPreservedBinSample`). On a very quiet segment the waterfall renders down to −255 while the derived spectrum line and squelch floor clamp at −200, so the two disagree. Either re-clamp for those consumers or lower `kKiwiSdrWaterfallMinDbm` to match. (#4069 review)

File: src/gui/MainWindow_KiwiSdr.cpp [hunk 6]
```diff
@@ -1489,6 +1514,36 @@ void MainWindow::wireKiwiSdr()
...
         });
+        connect(m_kiwiSdrManager, &KiwiSdrManager::waterfallDisplayRangeChanged,
+                this, [this](const QString& profileId, float minDbm,
+                             float maxDbm, bool serverAuto) {
+            if (profileId.isEmpty() || !m_kiwiSdrManager) {
+                return;
+            }
+
+            const int sliceId =
+                m_kiwiSdrManager->assignedSliceForProfile(profileId);
```
Comment (ten9876) [inline src/gui/MainWindow_KiwiSdr.cpp]: **Auto range never reaches the sliders/pan for tracking-only-fallback profiles.** This handler resolves the slice via `assignedSliceForProfile(profileId)` and early-returns if it's < 0. A profile in the documented owned-but-unassigned fallback (`connectProfile` → `profileNeedsInitialTracking` → `primeProfileTracking` → `setTrackedSlice`, but never `assignSliceToProfile`, so `m_sliceAssignments` has no entry) renders a waterfall and computes an Auto range on the client, but this early-returns — so `setKiwiSdrWaterfallDisplayRange` never fires and the WF Floor/Ceiling sliders never move even though Auto succeeded. Consider resolving the target pan via the profile→pan mapping rather than requiring an assigned slice. (#4069 review)

File: src/gui/SpectrumOverlayMenu.cpp [hunk 7]
```diff
@@ -82,7 +83,7 @@ static constexpr int kKiwiSdrWaterfallRateMax = 4;
 static QString kiwiWaterfallDbText(int db)
 {
     return QStringLiteral("%1%2 dB")
```
Comment (ten9876) [inline src/gui/SpectrumOverlayMenu.cpp]: `kiwiWaterfallDbText` still carries the old signed-offset formatting (a `+` for positives and a ` dB` suffix), but in Kiwi mode the WF Floor/Ceiling slider values are now *absolute dBm* boundaries (e.g. −110, −10), not −30..+30 offsets. So the slider reads “−110 dB” for what is a −110 **dBm** power boundary, and the `+` branch never fires for the normal all-negative range. Suggest ` dBm` (and dropping the now-dead `+` prefix) for the Kiwi labels — confirm the helper isn't shared with a genuinely-relative dB display first. (#4069 review)

**Comment by rfoust** [comment]
Pushed follow-up cleanup in rfoust/AetherSDR@2c23242d.

What changed:
- Aligned the Kiwi waterfall fallback floor with the protocol decode (`value - 255`) by using `-255 dBm` instead of `-200 dBm`.
- Corrected the clean-room WF Rate notes to the implemented compact `0..4` / fixed `1..4` range.
- Fixed the Display panel readout column so `WF Floor` / `WF Ceiling` dBm labels fit and do not relayout the other slider rows while dragging.

Validation:
- `git diff --check`
- `cmake --build build --target AetherSDR --parallel`
- `cmake --build build --target kiwi_sdr_protocol_test --parallel`
- `./build/kiwi_sdr_protocol_test`

File: src/core/KiwiSdrClient.cpp (hunk 1 above)
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp:1567]: Fixed in the amend — the new `clampedWaterfallDisplayRange()` is applied inside `adjustedAutoWaterfallDisplayRange` (both auto branches) and in `updateWaterfallAutoScale` before storing, so the Auto range is now clamped to the same `[-260, 30]` limits as the manual path. Verified. Resolving.

File: src/core/KiwiSdrManager.cpp (hunk 2 above)
Comment (ten9876) [reply src/core/KiwiSdrManager.cpp:894]: Fixed in the amend — legacy profiles now migrate to Auto (defaults + `waterfallAutoScale = true`) rather than re-baselining the offsets. Verified. Resolving.

File: src/core/KiwiSdrClient.cpp (hunk 3 above)
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp:1622]: Fixed in the amend — a `wf_cal` change now calls `resetWaterfallAutoScaleHistory()` (and re-arms pending), so the 20-row window can't mix differently-calibrated rows. Verified. Resolving.

File: src/core/KiwiSdrClient.cpp (hunk 4 above)
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp]: Fixed in the amend — the guard is now `(!m_trackedBandName.isEmpty() || !normalizedBandName.isEmpty()) && old != new`, so an unknown→known band transition resets the Auto history. Verified. Resolving.

File: src/core/KiwiSdrProtocol.cpp (hunk 5 above)
Comment (ten9876) [reply src/core/KiwiSdrProtocol.cpp:497]: Fixed in the amend — `kKiwiSdrWaterfallMinDbm` was lowered from -200 to -255 dBm to match the expanded `waterfallByteToDbm` range, so the FFT-trace/squelch consumers and the waterfall now share the same floor. Verified. Resolving.

File: src/gui/MainWindow_KiwiSdr.cpp (hunk 6 above)
Comment (ten9876) [reply src/gui/MainWindow_KiwiSdr.cpp]: Addressed in b68940a6 (maintainer fix). The handler now resolves target pane(s) via `kiwiSdrProfileForPan` — the same resolution the display/sync path uses — instead of `assignedSliceForProfile`, so the computed Auto range reaches whichever pane is actually displaying the profile (including diversity/active-slice resolution and the owned-but-unassigned tracking fallback), and updates every pane when the profile is shown on more than one. Builds clean. Resolving.

File: src/gui/SpectrumOverlayMenu.cpp (hunk 7 above)
Comment (ten9876) [reply src/gui/SpectrumOverlayMenu.cpp]: Fixed in the amend — `kiwiWaterfallDbText` now emits " dBm" (absolute), and the Stabilize commit added a value-column width reservation so `-260 dBm` fits without relayout during drag. Verified. Resolving.

**Review by ten9876 — APPROVED** [review]
Approving. Every finding from the review is now resolved, and the feature is in good shape.

**Your fixes (verified against the amended code):**
- Auto floor/ceiling are now clamped to `[-260, 30]` via the new `clampedWaterfallDisplayRange()`, applied inside `adjustedAutoWaterfallDisplayRange` and in `updateWaterfallAutoScale` before storing — so both auto branches are covered.
- Legacy profiles migrate to Auto instead of re-baselining stale offsets.
- A `wf_cal` change now resets the auto-scale history (nice refactor into the shared `resetWaterfallAutoScaleHistory()` helper), so the percentile window can't mix calibrations.
- The band-change reset now triggers on unknown→known transitions.
- The `waterfallByteToDbm` floor (−255) and the consumer floor (`kKiwiSdrWaterfallMinDbm`) are aligned.
- WF Floor/Ceiling labels read `dBm`, with a column-width reservation so they don't relayout during drag.

The new helpers are correct — `clampedWaterfallDisplayRange` clamps max against the already-clamped min (no inverted range), and the reset helper properly invalidates and re-arms.

**One maintainer fix pushed** (b68940a6) for the last item — the `waterfallDisplayRangeChanged` handler resolved its pan via `assignedSliceForProfile` and could drop the computed range for a pan displaying the profile through diversity/active-slice resolution or the owned-but-unassigned tracking fallback (and only updated one pan). It now resolves pane(s) via `kiwiSdrProfileForPan` — the same path the display/sync code uses — and updates every pane showing the profile.

Verified locally: clean build (751/751, 0 warnings in the touched files) and `kiwi_sdr_protocol_test` green, on the branch rebased onto current `main` (with `HAVE_PIPEWIRE_NATIVE` defined here). The 2D texture path + clean-room percentile Auto are a solid improvement. Approving and enabling auto-merge. 73 — reviewed with Claude Code on behalf of KK7GWY.

File: src/core/KiwiSdrClient.cpp [hunk 8]
```diff
@@ -2336,6 +2562,61 @@ void KiwiSdrClient::handleTextMessage(StreamKind stream, const QString& text)
...
+        }
+
+        if (token.key == QStringLiteral("wf_cal")) {
+            const int calibrationDb = std::clamp(static_cast<int>(std::lround(value)),
+                                                 -200,
+                                                 200);
+            if (m_waterfallCalibrationDb != calibrationDb) {
+                m_waterfallCalibrationDb = calibrationDb;
+            }
+        } else if (token.key == QStringLiteral("mindb")) {
```
[the remark this answers is not in the document]
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp:2621]: Maintainer note: this is already addressed in the current head — `handleTextMessage` now calls `resetWaterfallAutoScaleHistory()` when `m_waterfallCalibrationDb` changes (KiwiSdrClient.cpp:2614-2616), which clears `m_waterfallAutoScaleRows` and invalidates the client auto range. This finding was against an earlier commit. Resolving.

File: src/core/KiwiSdrClient.cpp [hunk 9]
```diff
@@ -1438,21 +1507,159 @@ void KiwiSdrClient::sendWaterfallViewToServer()
...
+        && std::abs(m_lastEmittedWaterfallMinDbm - range.minDbm) < 0.05f
+        && std::abs(m_lastEmittedWaterfallMaxDbm - range.maxDbm) < 0.05f) {
+        return;
+    }
+
+    m_lastEmittedWaterfallMinDbm = range.minDbm;
+    m_lastEmittedWaterfallMaxDbm = range.maxDbm;
+    m_lastEmittedWaterfallRangeValid = true;
+    m_lastEmittedWaterfallServerAuto = autoRange;
+    emit waterfallDisplayRangeChanged(range.minDbm, range.maxDbm, autoRange);
```
[the remark this answers is not in the document]
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp:1613]: Maintainer note: functionally correct — the flag means "an auto range is active" (client- or server-sourced), which is exactly what the overlay UI consumes to show Auto as checked; no consumer needs to distinguish the two sources. The `serverAuto` parameter name on the signal is a cosmetic misnomer worth a rename, but there's no behavioral defect. Non-blocking. Resolving.

File: src/core/KiwiSdrClient.cpp [hunk 10]
```diff
@@ -1438,21 +1507,159 @@ void KiwiSdrClient::sendWaterfallViewToServer()
...
+        && std::abs(m_waterfallClientAutoMinDbm - range.minDbm)
+            <= kWaterfallAutoReuseToleranceDb
+        && std::abs(m_waterfallClientAutoMaxDbm - range.maxDbm)
+            <= kWaterfallAutoReuseToleranceDb) {
+        m_waterfallAutoScalePending = false;
+        if (force) {
+            sendWaterfallDisplayAdjustmentsToServer();
+            emitWaterfallDisplayRangeChanged(true);
+        }
+        return;
```
[the remark this answers is not in the document]
Comment (ten9876) [reply src/core/KiwiSdrClient.cpp:1670]: Maintainer note: accurate — the inner `if (force)` is unreachable dead code (the intended always-publish-on-force behavior is achieved by the fall-through, so no functional impact). Worth deleting in a cleanup pass but non-blocking. Resolving.

File: src/gui/SpectrumWidget.cpp [hunk 11]
```diff
@@ -3609,6 +3606,14 @@ void SpectrumWidget::clearCurrentWaterfallRows()
 #endif
 }
 
+void SpectrumWidget::resetKiwiSdrWaterfallDisplayRange()
+{
+    m_kiwiSdrDisplayFloorDbm = -110.0f;
+    m_kiwiSdrDisplayCeilDbm = -10.0f;
+    m_kiwiSdrDisplayRangeValid = false;
+    m_kiwiSdrDisplayRangeServerAuto = false;
+}
```
[the remark this answers is not in the document]
Comment (ten9876) [reply src/gui/SpectrumWidget.cpp:3617]: Maintainer note: accurate and matches an item from my own review — `resetKiwiSdrWaterfallDisplayRange()` should source the default floor/ceiling from `KiwiSdrProtocol::defaultWaterfallDisplayRange()` rather than hardcoding -110/-10. Minor maintainability nit (the literals happen to match the protocol defaults today), non-blocking follow-up. Resolving.

File: docs/kiwisdr-cleanroom-design.md [hunk 12]
```diff
@@ -48,7 +48,7 @@ black-box observations made in this thread.
   local "Not connected to KiwiSDR" overlay with the connection detail until the
   profile reconnects; Settings -> Antennas should show each Kiwi profile as a
   readable name/server entry; and the panadapter waterfall controls should map
-  to Kiwi W/F cell, floor, and rate while a Kiwi antenna is selected, then
+  to Kiwi W/F gain, floor, and rate while a Kiwi antenna is selected, then
   return to normal Flex waterfall behavior when it is not.
```
[the remark this answers is not in the document]
Comment (ten9876) [reply docs/kiwisdr-cleanroom-design.md:52]: Maintainer note: accurate doc nit — the on-screen controls are WF Ceiling / WF Floor / WF Rate, so "gain, floor, and rate" in this section is stale wording. Non-blocking doc follow-up. Resolving.
